sdk-core/transaction-intents-factories/smart_contract_transaction_intents_factory.md

```diff
@@ -5,12 +5,15 @@ class SmartContractTransactionIntentsFactory:
     // The constructor is not captured by the specs; it's up to the implementing library to define it.
     // Generally speaking, the constructor should be parametrized with a configuration object which defines entries such as:
     // "minGasLimit", "gasLimitPerByte" etc.
-    // The constructor should also be parametrized with a Codec instance, necessary to encode contract call arguments.
+    // The constructor may also be parametrized with a Codec instance, necessary to encode contract call arguments.
 
     create_transaction_intent_for_deploy({
         sender: IAddress;
         bytecode: bytes OR bytecodePath: Path;
         arguments: List[object] = [];
+        // Question for review: is this a valid case?
```

Comment:
I've tested this with a contract that was payable on the `init` method, and it works. Indeed, this is a valid case. Same for the `upgrade` method.

Reply:
Good, thanks :tada: :raised_hands: